File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/appContext/UserMetadataOthers.java
package com.pubnub.docs.appContext;

import com.pubnub.api.PubNubException;
import com.pubnub.api.java.PubNub;
import com.pubnub.api.java.models.consumer.objects_api.uuid.PNSetUUIDMetadataResult;
import com.pubnub.docs.SnippetBase;

public class UserMetadataOthers extends SnippetBase {
    private void getUUIDMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-1

        PubNub pubNub = createPubNub();

        // snippet.getUUIDMetadataBasic
        pubNub.getUUIDMetadata().async(result -> { /* check result */ });
        // snippet.end
    }

    private void setUUIDMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-2

        PubNub pubNub = createPubNub();

        // snippet.setUUIDMetadataBasic
        PNSetUUIDMetadataResult pnSetUUIDMetadataResult = pubNub.setUUIDMetadata()
                .name("Foo")
                .profileUrl("http://example.com")
                .email("foo@example.com")
                .includeCustom(true)
                .sync();
        // snippet.end
    }

    private void removeUUIDMetadataBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/objects#basic-usage-3

        PubNub pubNub = createPubNub();

        // snippet.removeUUIDMetadataBasic
        pubNub.removeUUIDMetadata().async(result -> { /* check result */ });
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/appContext/UserMetadataOthers.kt
package com.pubnub.docs.appContext

import com.pubnub.api.PubNub

class UserMetadataOthers {
    private fun getUUIDMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-1

        // snippet.getUUIDMetadataBasic
        pubnub.getUUIDMetadata()
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }

    private fun setUUIDMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-2

        // snippet.setUUIDMetadataBasic
        pubnub.setUUIDMetadata()
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }

    private fun removeUUIDMetadataBasic(pubnub: PubNub) {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/objects#basic-usage-3

        // snippet.removeUUIDMetadataBasic
        pubnub.removeUUIDMetadata()
            .async { result ->
                result.onFailure { exception ->
                    // Handle error
                }.onSuccess { value ->
                    // Handle successful method result
                }
            }
        // snippet.end
    }
}
